plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-calculator/shipping-calculator.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { useContext } from '@wordpress/element';
+import type { ShippingAddress } from '@woocommerce/settings';
+import { useCustomerData } from '@woocommerce/base-context/hooks';
+import { dispatch } from '@wordpress/data';
+import { CART_STORE_KEY, processErrorResponse } from '@woocommerce/block-data';
+import { StoreNoticesContainer } from '@woocommerce/blocks-components';
+import { removeNoticesWithContext } from '@woocommerce/base-utils';
+
+/**
+ * Internal dependencies
+ */
+import ShippingCalculatorAddress from './address';
+import { ShippingCalculatorContext } from './context';
+import './style.scss';
+
+interface ShippingCalculatorProps {
+	onUpdate?: ( newAddress: ShippingAddress ) => void;
+	onCancel?: () => void;
+	addressFields?: Partial< keyof ShippingAddress >[];
+}
+
+export const ShippingCalculator = ( {
+	onUpdate = () => {
+		/* Do nothing */
+	},
+	onCancel = () => {
+		/* Do nothing */
+	},
+	addressFields = [ 'country', 'state', 'city', 'postcode' ],
+}: ShippingCalculatorProps ): JSX.Element | null => {
+	const {
+		shippingCalculatorID,
+		showCalculator,
+		setIsShippingCalculatorOpen,
+	} = useContext( ShippingCalculatorContext );
+	const { shippingAddress } = useCustomerData();
+	const noticeContext = 'wc/cart/shipping-calculator';
+
+	if ( ! showCalculator ) {
+		return null;
+	}
+
+	return (
+		<div
+			className="wc-block-components-shipping-calculator"
+			id={ shippingCalculatorID }
+		>
+			<StoreNoticesContainer context={ noticeContext } />
+			<ShippingCalculatorAddress
+				address={ shippingAddress }
+				addressFields={ addressFields }
+				onCancel={ () => {
+					setIsShippingCalculatorOpen( false );
+					onCancel();
+				} }
+				onUpdate={ ( newAddress ) => {
```

Comment:
Hey @mikejolley, this might be a dumb question, but I've seen this around and was wondering: Would this cause the function to be redefined on each render? I know I might be talking peanuts here, but would defining an update handler outside be better in terms of performance?

Well, it's just a question, I'm approving it everything's working nicely :)